### PR TITLE
protobuf: add 3.19.1 + modernize

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

2 comments:

- Since we wrap upstream CMakeLists with our CMakeLists (it defeats any CMake policy set by upstream), the patch for 3.19.0 (https://github.com/protocolbuffers/protobuf/pull/9153) is not really required for the moment. But il will be necessary with CMakeToolchain generator.

- when cross-building test package is broken if shared, at least on Macos, because protobuf is also in build requirements and when calling protoc, shared lib of build context is not found or the one of host context is found. I don't have a fix...

closes https://github.com/conan-io/conan-center-index/issues/8534

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by killzoner** [comment]
> It's explained in the second comment [#7857 (comment)](https://github.com/conan-io/conan-center-index/pull/7857#issue-767787378)
> 
> You can raise `ConanInvalidConfiguration` in `validate()` if shared and Macos and armv8, but it's not very neat. The build succeeds, but test package is broken in this case.

Actually it might be an option as it's officially stated that apple M1is officially unsupported, support for ARMV8 is only on linux as of now (https://grpc.io/blog/grpc-on-arm64/).

File: recipes/protobuf/all/conanfile.py [hunk 1]
```diff
@@ -31,10 +32,14 @@ class ProtobufConan(ConanFile):
 
     short_paths = True
 
-    exports_sources = ["CMakeLists.txt", "patches/*"]
     generators = "cmake"
     _cmake = None
 
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
```
[the remark this answers is not in the document]
Comment (SpaceIm) [reply recipes/protobuf/all/conanfile.py:40]: New best practice I would say. See https://github.com/conan-io/conan-center-index/issues/7790

**Comment by SpaceIm** [comment]
So I drop cross-build if shared and Macos to move forward, until we can find a workaround for CMake build in test package (the issue is that when protoc of build context is called, it doesn't find the shared lib in `DYLD_LIBRARY_PATH` since SIP clears this env variable).

**Review by ericriff — DISMISSED** [review]

**Comment by SpaceIm** [comment]
@uilianries @SSE4 @prince-chrismc @madebr @ericLemanissierBot Could you review please? It provides protobuf static on macOS M1 to move forward on other recipes.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by SSE4 — APPROVED** [review]

**Review by madebr — APPROVED** [review]

File: recipes/protobuf/all/conanfile.py [hunk 2]
```diff
@@ -219,27 +238,21 @@ def package_info(self):
...
-
+        # libprotobuf-lite
         if self.options.lite:
-            self.cpp_info.components["libprotobuf-lite"].names["cmake_find_package"] = "libprotobuf-lite"
-            self.cpp_info.components["libprotobuf-lite"].names["cmake_find_package_multi"] = "libprotobuf-lite"
-            self.cpp_info.components["libprotobuf-lite"].names["pkg_config"] = "protobuf-lite"
+            self.cpp_info.components["libprotobuf-lite"].set_property("cmake_target_name", "protobuf::libprotobuf-lite")
+            self.cpp_info.components["libprotobuf-lite"].set_property("pkg_config_name", "protobuf-lite")
+            self.cpp_info.components["libprotobuf-lite"].builddirs.append(self._cmake_install_base_path)
+            self.cpp_info.components["libprotobuf-lite"].set_property("cmake_build_modules", build_modules)
```
Comment (SpaceIm) [inline recipes/protobuf/all/conanfile.py:252]: should be define at global scope since 1.43.2: https://github.com/conan-io/conan/issues/10282

**Pull request merged**